### PR TITLE
Fix for many optional query parameters

### PR DESCRIPTION
**Pull request by gregor-rayman**
Solves #2942

### CODE REVIEW

**Comment by gregor-rayman** [comment]
Added alternative encoding of query parameters. Instead of `query[String](query("name")).optional` which doubles the number of handlers, `query[Option[String]](queryOptional("name")` can read the optional query parameter directly without the need of a Fallback handler.

File: zio-http/shared/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala [hunk 1]
```diff
@@ -287,9 +287,10 @@ private[codec] object EncoderDecoder {
 
         val params = queryParams.queryParamsOrElse(query.name, Nil)
 
-        if (params.isEmpty)
-          throw HttpCodecError.MissingQueryParam(query.name)
-        else {
+//        if (params.isEmpty)
```
Comment (jdegoes) [inline zio-http/shared/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala]: This code is commented out. Why?

**Comment by jdegoes** [comment]
Closed in favor of @987Nabil 's solution, which allow's deserializing any number of query parameters into a case class.

**Comment by gregor-rayman** [comment]
Great! His solution is much better

**Pull request closed without merging**